### PR TITLE
Add GLUE dataset (but still one issue with QQP and SNLI, see comment)

### PR DESCRIPTION
**Pull request by PattynR**
There is one remaining issue that lies with the dataset itself. The code works for all the datasets except for SNLI and QQP where there are some lines in the data file that contain too much or too little data fields. See my comment in the issue discussion for more explanations. I think that someone that is used to those datasets should be able to know what to do.

### CODE REVIEW

**Comment by PetrochukM** [comment]
Hey There! Sorry for not responding in a while, please update the tests folder to test your changes!

**Comment by PattynR** [comment]
Hey, no problem! I've added a test file and I've used flake8 to re-format my files, I however have an issue when I try to install Yapf on my computer, but I've run an online PEP8 analyser, and the only warnings left are for too long lines, so I think it should be ok.

**Comment by PattynR** [comment]
Hey, do I have to open a new pull request, or can you see my changes in this ones? It's the first time I do a pull request so I don't really now how things should go.

**Comment by PetrochukM** [comment]
Hi! Can you make sure that travis-ci is happy, please?

**Comment by PetrochukM** [comment]
Also, looked over your code. The module you wrote is gigantic, please do not create a ["god module"](https://en.wikipedia.org/wiki/God_object)

File: torchnlp/datasets/glue.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                        pass
+            ret.append(Dataset(examples))
+
+    if len(ret) == 1:
+        return ret[0]
+    else:
+        return tuple(ret)
+
+
+def mrpc_processing(folder_path, mrpc_train_file, mrpc_test_file):
```
Comment (PetrochukM) [inline torchnlp/datasets/glue.py]: There is no comments on this function, please document it. It's hard to understand what it's doing.

File: torchnlp/datasets/glue.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          'sentence': 'Immediately behind the basilica ... ',
+          'label': 'entailment'
+        }, {
+          'index': '1',
+          'question': 'What is the Grotto at Notre Dame?',
+          'sentence': 'It is a replica of the grotto at ... ',
+          'label': 'not_entailment'
+        }]
+    """
+    if dataset is None:
```
Comment (PetrochukM) [inline torchnlp/datasets/glue.py]: This module is crazy large, please break it up.

File: torchnlp/datasets/glue.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    else:
+        for filename in splits:
+            examples = []
+            with open(os.path.join(folder_path, filename), newline='') as tsvfile:
+                tsvreader = csv.reader(tsvfile, delimiter='\t', quoting=csv.QUOTE_NONE)
+                keys = next(tsvreader)
+                nb_arguments = len(keys)
+                for line in tsvreader:
+                    """
+                        problem with QQP and SNLI : len(line) is different from nb_arguments for some lines ...
```
Comment (PetrochukM) [inline torchnlp/datasets/glue.py]: Why are you using a multiline comment here?

**Comment by PattynR** [comment]
Ok I'll work on it.

**Pull request closed without merging**